controllers/helpers.go

```diff
@@ -68,6 +68,10 @@ const (
 	deprecatedManagerCredsWarning = "You're using deprecated functionality: " +
 		"Using Azure credentials from the manager environment is deprecated and will be removed in future releases. " +
 		"Please specify an AzureClusterIdentity for the AzureCluster instead, see: https://capz.sigs.k8s.io/topics/multitenancy.html "
+	// ToDo: Find a way to make this configurable for a user.
+	// This is the path where the projected service account token should be present for
+	// cloud provider azure.
+	aadFederatedTokenFilePath = "/var/run/secrets/azure/tokens/azure-identity-token" //nolint:gosec // Path of projected service account token
```

Comment:
I think instead of making an API change to configure this. I think we should first go with using an annotation in `AzureClusterIdentity` object. 
@CecileRobertMichon @mboersma -- thoughts??

Reply:
How does this file get on the VM?